diff --git a/dnnlib/util.py b/dnnlib/util.py
index 4abbbf2..4a1c7b1 100755
--- a/dnnlib/util.py
+++ b/dnnlib/util.py
@@ -29,10 +29,18 @@ import urllib
 import urllib.request
 import uuid
 
-from distutils.util import strtobool
 from typing import Any, List, Tuple, Union
 
 
+def strtobool(value: str) -> int:
+    value = value.lower()
+    if value in ("y", "yes", "t", "true", "on", "1"):
+        return 1
+    if value in ("n", "no", "f", "false", "off", "0"):
+        return 0
+    raise ValueError("invalid truth value {!r}".format(value))
+
+
 # Util classes
 # ------------------------------------------------------------------------------------------
 
