diff --git a/training/training_loop.py b/training/training_loop.py
index 14836ad..e4c77b5 100755
--- a/training/training_loop.py
+++ b/training/training_loop.py
@@ -357,7 +357,7 @@ def training_loop(
             for name, module in [('G', G), ('D', D), ('G_ema', G_ema), ('augment_pipe', augment_pipe)]:
                 if module is not None:
                     if num_gpus > 1:
-                        misc.check_ddp_consistency(module, ignore_regex=r'.*\.w_avg')
+                        misc.check_ddp_consistency(module, ignore_regex=r'.*\.(w_avg|noise_const)')
                     module = copy.deepcopy(module).eval().requires_grad_(False).cpu()
                 snapshot_data[name] = module
                 del module # conserve memory
